db/template_test.go

```diff
@@ -248,6 +239,118 @@ func TestDeleteTemplate(t *testing.T) {
 	}
 }
 
+func TestGetTemplate(t *testing.T) {
+	ctx := context.Background()
+	tests := []struct {
+		// Name identifies the single test in a table test scenario
+		Name string
+		// Input is a list of workflows that will be used to pre-populate the database
+		Input []*workflow.Workflow
+		// InputAsync if set to true inserts all the input concurrently
+		InputAsync bool
+	}{
+		{
+			Name: "get-template",
+			Input: []*workflow.Workflow{
+				func() *workflow.Workflow {
+					w := workflow.MustParseFromFile("./testdata/template_happy_path_1.yaml")
+					w.ID = "545f7ce9-5313-49c6-8704-0ed98814f1f7"
+					return w
+				}(),
+			},
+		},
+		{
+			Name:       "stress-get-template",
+			InputAsync: true,
+			Input: func() []*workflow.Workflow {
+				input := []*workflow.Workflow{}
+				for i := 0; i < 20; i++ {
+					w := workflow.MustParseFromFile("./testdata/template_happy_path_1.yaml")
+					w.ID = uuid.New().String()
+					w.Name = fmt.Sprintf("id_%d", rand.Int())
+					t.Log(w.Name)
+					input = append(input, w)
+				}
+				return input
+			}(),
+		},
+	}
+	for _, s := range tests {
+		t.Run(s.Name, func(t *testing.T) {
+			t.Parallel()
+			_, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{
+				ApplyMigration: true,
+			})
+			defer func() {
+				err := cl()
+				if err != nil {
+					t.Error(err)
+				}
+			}()
+
+			for _, in := range s.Input {
+				err := createTemplateFromWorkflowType(ctx, tinkDB, in)
+				if err != nil {
+					t.Error(err)
+				}
+			}
+
+			var wg sync.WaitGroup
+			wg.Add(len(s.Input))
+			for _, in := range s.Input {
+				if s.InputAsync {
+					go func(t *testing.T, ctx context.Context, tt *workflow.Workflow, tinkDB *db.TinkDB) {
```

Comment:
Same as before, it does not have sense to use `InputAsync` here because you are not doing any input.